src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/reference/inputs/EceCharacterReferenceContact.vue

```diff
@@ -0,0 +1,182 @@
+            maxlength="25"
+            hide-details="auto"
+            @update:model-value="(value) => $emit('update:model-value', value, { certificateNumber: value })"
+            @keypress="isNumber($event)"
+          ></v-text-field>
+        </v-col>
+      </v-row>
+      <v-row>
+        <v-col cols="12" md="8" lg="6" xl="4">
+          <v-text-field
+            v-if="certificateProvinceId && !userSelectProvinceIdBC"
+            v-model="dateOfBirth"
+            label="Your date of birth (Optional)"
+            variant="outlined"
+            color="primary"
+            hide-details="auto"
+            type="date"
+            :max="today"
+            @update:model-value="(value) => $emit('update:model-value', value, { dateOfBirth: value })"
+          ></v-text-field>
+        </v-col>
+      </v-row>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { DateTime } from "luxon";
+import { defineComponent } from "vue";
+import type { VTextField } from "vuetify/components";
+
+import type { FormData } from "@/store/form";
+import { useWizardStore } from "@/store/wizard";
+import { formatDate } from "@/utils/format";
+import { isNumber } from "@/utils/formInput";
+import * as Rules from "@/utils/formRules";
+
+export default defineComponent({
+  name: "EceCharacterReferenceContact",
+  components: {},
+  emits: {
+    "update:model-value": (_value: any, _updateFormData?: FormData) => true,
+  },
+  setup: () => {
+    const wizardStore = useWizardStore();
+    return { wizardStore };
+  },
+  data() {
+    return {
+      selection: undefined,
+      Rules,
+      certificateProvinceId: "",
+      dateOfBirth: "",
+      certificateNumber: "",
+    };
+  },
+  computed: {
+    provinces() {
+      return [
+        { title: "British Columbia", value: "BC" },
+        { title: "Alberta", value: "AB" },
+        { title: "Prince Edward Island", value: "PEI" },
+      ];
+    },
+    userSelectProvinceIdBC(): boolean {
+      return this.certificateProvinceId === "BC";
+    },
+    today() {
+      return formatDate(DateTime.now().toString());
+    },
+  },
+  methods: {
+    isNumber,
+    customOptionalIfNotBCRule() {
+      if (this.userSelectProvinceIdBC) {
+        return (v: string) => !!(v && v?.trim()) || "Required";
+      }
+      return true;
+    },
+    certificateProvinceIdChanged(value: string) {
+      (this.$refs.certificateNumberRef as VTextField)?.resetValidation();
+      if (value === "BC") {
+        this.$emit("update:model-value", value, { dateOfBirth: "", certificateProvinceId: value });
+        this.dateOfBirth = "";
+      } else {
+        this.$emit("update:model-value", value, { certificateProvinceId: value });
+      }
+    },
+    provinceClearClicked() {
+      this.certificateNumber = "";
+      this.dateOfBirth = "";
+      this.certificateProvinceId = "";
+      this.$emit("update:model-value", "", { dateOfBirth: "", certificateNumber: "", certificateProvinceId: "" });
```

Comment:
@SoLetsDev I think we may be over emitting with this setup, my understanding is that when you change the data values that are modelled by our input components:
```
      this.certificateNumber = "";
      this.dateOfBirth = "";
      this.certificateProvinceId = "";
```
This triggers an emit of `update:model-value` for each of the fields. Then calling `this.$emit("update:model-value", "", { dateOfBirth: "", certificateNumber: "", certificateProvinceId: "" }); ` would potentially be emitting again...

Reply:
Strangely it doesn't seem to fire off when we set the data props to "" when it has a value.